### PR TITLE
Am 9

### PR DESCRIPTION
**Pull request by YuliyaKVAS**

### CODE REVIEW

**Comment by lessarea** [comment]
could you please remove json-server from this project to keep it clearer? you can start it from other place

File: src/js/api.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  return instance.post(url, body, options);
+};
+
+export const putData = (url, body) => instance.put(url, body);
+
+export const putById = (url, id, body) => instance.put(`${url}/${id}`, body);
+
+export const deleteData = (url, id) => instance.delete(`${url}/${id}`);
+
+export default {
```
Comment (lessarea) [inline src/js/api.js]: it's not necessary, you already exported them via 'export'

File: src/js/components/CoursesList/CoursesList.js [hunk 2]
```diff
@@ -1,26 +1,46 @@
...
-    </Button>
-  </div>
-);
+  componentDidMount() {
+    getCourses()
+      .then(courses => this.setState({ courses }));
+  }
+
+  render() {
+    return (
```
Comment (lessarea) [inline src/js/components/CoursesList/CoursesList.js]: what is displayed when the date is loaded?
maybe show some kind of loader or nothing a good idea, what do you think?

File: src/js/components/Header/Header.js [hunk 3]
```diff
@@ -1,32 +1,41 @@
...
-    <div className={styles.logoName}>
-      <div className={styles.logo}>
-        Logo
+const Header = () => {
+  const handleLogout = () => {
+    localStorage.removeItem('token');
+  };
+  const loginLink = props => <Link to="/login" {...props} />;
+  return (
+    <div className={styles.headerWrapper}>
```
Comment (lessarea) [inline src/js/components/Header/Header.js]: you know what to do with this infinite duplications) i'm talking about 'styles....'

File: src/js/components/Header/Header.js [hunk 4]
```diff
@@ -1,32 +1,41 @@
...
-const Header = () => (
-  <div className={styles.headerWrapper}>
-    <div className={styles.logoName}>
-      <div className={styles.logo}>
-        Logo
+const Header = () => {
+  const handleLogout = () => {
+    localStorage.removeItem('token');
+  };
+  const loginLink = props => <Link to="/login" {...props} />;
```
Comment (lessarea) [inline src/js/components/Header/Header.js]: if we have a method here, isn't it better to turn the function into a class?

File: src/js/components/ListItem/ListItem.js [hunk 5]
```diff
@@ -5,42 +5,41 @@ import style from './ListItem.less';
...
-        </div>
-      </div>
-      <div className={style.second}>
-        <span>
-          {props.text}
+const ListItem = ({
+  title, timing, date, text, classes
+}) => (
+  <Paper className={classes.root}>
+    <div className={style.first}>
```
Comment (lessarea) [inline src/js/components/ListItem/ListItem.js]: duplications again and you import styles but get properties from style? is it work correctly?

File: src/js/components/LoginPage/LoginPage.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import TextField from '../TextField';
+import Button from '../Button';
+import styles from './LoginPage.less';
+
+const coursesLink = props => <Link to="/courses" {...props} />;
```
Comment (lessarea) [inline src/js/components/LoginPage/LoginPage.js]: in another component you create smth similar using a class method, please stick to one style

File: src/js/components/LoginPage/LoginPage.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+class LoginPage extends React.PureComponent {
+  state = {
+    email: '',
+    password: ''
+  };
+
+  handleChangeFieldValue = (event, field) => {
+    this.setState({ [field]: event.target.value });
```
Comment (lessarea) [inline src/js/components/LoginPage/LoginPage.js]: you can get this field value using name textfield attribute and then write 
const { name, value } = event.target;
this.setState({ [name]: value });

File: src/js/helpers.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+export const parseSecToMinutes = (sec) => {
+  const minutes = Math.floor(sec / 60);
+  const seconds = sec % 60;
+  return minutes ? `${minutes}m ${seconds}s` : `${sec}s`;
```
Comment (lessarea) [inline src/js/helpers.js]: what if that 12200000 seconds?)

File: src/js/components/SearchPanel/SearchPanel.js [hunk 9]
```diff
@@ -1,29 +1,21 @@
 import React from 'react';
-import { withStyles } from '@material-ui/core/styles';
-import TextField from '@material-ui/core/TextField';
+import TextField from '../TextField';
 import style from './SearchPanel.less';
 import Button from '../Button';
-import styles from './styles';
+
+/* eslint-disable */
```
Comment (lessarea) [inline src/js/components/SearchPanel/SearchPanel.js]: stop do that)) if some rules are bother you, just turn them off

File: src/js/components/ProtectedRoute/ProtectedRoute.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router-dom';
+
+/* eslint-disable */
+
+const ProtectedRoute = props => (
+  localStorage.getItem('token') ?
```
Comment (lessarea) [inline src/js/components/ProtectedRoute/ProtectedRoute.js]: what happened with code style in this file?)))

File: src/js/components/CoursesList/CoursesList.js (hunk 2 above)
Comment (YuliyaKVAS) [reply src/js/components/CoursesList/CoursesList.js]: Loader added)

File: src/js/helpers.js (hunk 8 above)
Comment (YuliyaKVAS) [reply src/js/helpers.js]: fixed

File: src/js/components/Header/Header.js (hunk 3 above)
Comment (YuliyaKVAS) [reply src/js/components/Header/Header.js]: I'm not exactly understand what do you mean about styles duplications..)

File: src/js/components/ListItem/ListItem.js (hunk 5 above)
Comment (YuliyaKVAS) [reply src/js/components/ListItem/ListItem.js]: here style and styles are different)
style is imported from ListItem.js, and styles from styles.js (this is mui styles)

File: src/js/components/Header/Header.js (hunk 3 above)
Comment (YuliyaKVAS) [reply src/js/components/Header/Header.js]: done)

File: src/js/components/ListItem/ListItem.js (hunk 5 above)
Comment (YuliyaKVAS) [reply src/js/components/ListItem/ListItem.js]: fixed)

File: src/js/components/CoursesList/CoursesList.js [hunk 11]
```diff
@@ -1,26 +1,52 @@
...
+      .then(() => this.setState({ isFetching: false }));
+  }
+
+  render() {
+    return (
+      <div className={styles.container}>
+        <span className={styles.header}>
+        Courses
+        </span>
+        {this.state.isFetching && <Loader />}
```
Comment (lessarea) [inline src/js/components/CoursesList/CoursesList.js]: destructuring is your friend. don't forget about that)
btw, isFetching return <Loader/> before return more obvious then && expression inside main return

File: src/js/components/Header/Header.js (hunk 3 above)
Comment (lessarea) [reply src/js/components/Header/Header.js]: ty)

File: src/js/components/CoursesList/CoursesList.js (hunk 11 above)
Comment (YuliyaKVAS) [reply src/js/components/CoursesList/CoursesList.js]: fixed)

**Pull request merged**